packages/cel/src/equals.test.ts

```diff
@@ -0,0 +1,276 @@
+// Copyright 2024-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { before, describe, test } from "node:test";
+import * as assert from "node:assert/strict";
+import {
+  CelList,
+  CelMap,
+  CelUint,
+  ProtoNull,
+  type CelVal,
+} from "./value/value.js";
+import { equals } from "./equals.js";
+import { getCelType } from "./value/type.js";
+import {
+  create,
+  createRegistry,
+  isMessage,
+  toJsonString,
+} from "@bufbuild/protobuf";
+import {
+  BoolValueSchema,
+  BytesValueSchema,
+  DoubleValueSchema,
+  DurationSchema,
+  FloatValueSchema,
+  Int32ValueSchema,
+  Int64ValueSchema,
+  StringValueSchema,
+  timestampFromMs,
+} from "@bufbuild/protobuf/wkt";
+import {
+  isReflectList,
+  isReflectMap,
+  isReflectMessage,
+  reflect,
+} from "@bufbuild/protobuf/reflect";
+import { CEL_ADAPTER } from "./adapter/cel.js";
+import * as type from "./value/type.js";
+import { TestAllTypesSchema } from "@bufbuild/cel-spec/cel/expr/conformance/proto2/test_all_types_pb.js";
+import { setEvalContext } from "./eval.js";
+
+/**
+ * The tests are based cases in this accepted CEL proposal: https://github.com/google/cel-spec/wiki/proposal-210#proposal
+ */
+describe("equals()", () => {
+  before(() =>
+    setEvalContext({
+      registry: createRegistry(TestAllTypesSchema),
+    }),
+  );
+  describe("must be true", () => {
+    const pairs = [
```

Comment:
There are some cases that the conformance test do not cover, others I've added to completeness and ease of testability.